Doc/library/asyncio-task.rst

```diff
@@ -537,6 +554,90 @@ Shielding From Cancellation
 Timeouts
 ========
 
+.. coroutinefunction:: timeout(delay)
+
+    A convenient way to limit the amount of time spent waiting on
+    something is to use the :func:`asyncio.timeout`
+    :ref:`asynchronous context manager <async-context-managers>`.
+
+    *delay* can either be ``None`` or a float or int number of
+    seconds to wait. If *delay* is ``None``, no time limit will
+    be applied. In either case, the context manager can be
+    rescheduled after creation.
+
+    Example::
+
+        async def main():
+            async with asyncio.timeout(10):
+                await long_running_task()
+
+    If ``long_running_task`` takes more than 10 seconds to complete,
+    the context manager will cancel the current task and handle
+    the resulting :exc:`asyncio.CancelledError` internally, transforming it
+    into a :exc:`asyncio.TimeoutError` which can be caught and handled.
+
+    .. note::
+
+      The :func:`asyncio.timeout` context manager is what transforms
+      the :exc:`asyncio.CancelledError` into an :exc:`asyncio.TimeoutError`,
+      which means the :exc:`asyncio.TimeoutError` can only be caught
+      *outside* of the context manager.
+
+    Example of catching :exc:`asyncio.TimeoutError`::
+
+        async def main():
+            try:
+                async with asyncio.timeout(10):
+                    await long_running_task()
+            except TimeoutError:
+                print("The long operation timed out, but we've handled it.")
+
+            print("This statement will run regardless.")
+
+    The context manager produced by :func:`asyncio.timeout` can be
+    rescheduled to a different deadline and inspected.
```

Comment:
This should use a new function heading so it's clear that this is about `reschedule` (and that gets added to the index). You may actually have to change the structure of the whole section to admit the existence of the `Timeout` class so you can document its methods (aren't there a few more worth documenting?)